src/services/fileStorage/docs/index.js

```diff
@@ -1,200 +1,895 @@
+						name: 'permissions',
+						type: 'Array',
+						style: 'matrix',
+						explode: true,
+						schema: {
+							type: 'array',
+							items: {
+								type: 'object',
+							},
+						},
+					},
+				],
+				description: '',
+				summary: 'Creates a new file object',
+				responses: {
+					200: {
+						description: 'Returns the data of the newly created file',
+						example: {
+							name: '24-1.gif',
+							owner: '0000dcfbfb5c7a3f00bf21ab',
+							type: 'image/gif',
+							size: 2545,
+							storageFileName: '1560977688023-24-1.gif',
+							thumbnail: 'https://schulcloud.org/images/login-right.png',
+							parent: '5c6490eeb07d3c7eacedc9a8',
+							refOwnerModel: 'course',
+							_id: '5d0aa11841ff829edda019c8',
+							updatedAt: '2019-05-29T10:54:48.987Z',
+							createdAt: '2019-05-29T10:54:48.987Z',
+							permissions: [
+								{
+									refId: '0000d231816abba584714c9e',
+									refPermModel: 'user',
+									delete: true,
+									create: true,
+									read: true,
+									write: true,
+								},
+								{
+									refId: '0000d186816abba584714c99',
+									refPermModel: 'role',
+									delete: false,
+									create: false,
+									read: true,
+									write: false,
+								},
+								{
+									refId: '0000d186816abba584714c98',
+									refPermModel: 'role',
+									delete: false,
+									create: false,
+									read: true,
+									write: true,
+								},
+							],
+							isDirectory: false,
+						},
+					},
+					403: {
+						description: 'File operation is forbidden due to missing permissions',
+					},
+					500: {
+						description: 'Error performing operation',
+					},
+				},
+			},
+			find: {
+				description: '',
+				summary: 'Retrieve all files and directories of given owner and parent folder',
+				parameters: [
+					{
+						in: 'query',
+						description: 'ID of either an user, course or team',
+						type: 'String (ObjectID)',
+						name: 'owner',
+						required: true,
+					},
+					{
+						in: 'query',
+						description: 'ID of parent folder',
+						type: 'String (ObjectID)',
+						name: 'parent',
+					},
+				],
+				responses: {
+					200: {
+						description: 'Returns an array of all files ',
+					},
+					403: {
+						description: 'File operation is forbidden due to missing permissions',
+					},
+					500: {
+						description: 'Error performing operation',
+					},
+				},
+			},
+			remove: {
+				parameters: [
+					{
+						description: 'File ID to be deleted',
+						required: true,
+						name: '_id',
+						in: 'query',
+						type: 'String (ObjectID)',
+					},
+					{
+						description: 'File ID to be deleted',
+						name: 'id',
+						in: 'path',
+						type: 'String (ObjectID)',
+					},
+				],
+				description: '',
+				summary: 'Removes the file with given Object ID either in query or path',
+				responses: {
+					200: {
+						description: 'OK - file deleted',
```

Comment:
normally on delete, the deleted entity hass to be included in the response, check json api for this and other methods

Reply:
Source?